source/slang/parser.cpp

```diff
@@ -990,7 +996,8 @@ namespace Slang
             case TokenType::QuestionMark:
                 if (AdvanceIf(parser, TokenType::Colon))
                 {
-                    nameToken.Content = nameToken.Content + ":";
+                    // Allocate a new slice, so we can concat : at the end
+                    nameToken.Content = parser->sink->sourceManager->allocateConcatStringSlice(nameToken.Content, UnownedStringSlice::fromLiteral(":"));
```

Comment:
In our approach, operator expressions are mostly just a different syntax for function calls, so that `a + b` is treated, after parsing, just like a call `+(a,b)` to simplify the rest of the compiler (a call like that will match a function declared as `operator+`, because the function will be registered with a name of `+`).

In the case here we are parsing an expression of the form `a ? b : c`, and the `nameToken` in this case is the `?` part. Rather than construct an expression like `?(a,b,c)`, we instead construct one of the form `?:(a,b,c)` which matches a declaration of `operator?:` in the standard library. (Aside: in the case of HLSL, the `?:` operation really does act like a function, since it does not short-circuit.)

So, the adding of `:` here is really just to turn `?` into `?:`, and that is really just so that we look up the callee function based on the correct name, so it might not even be important to modify the `Content` of the token, so long as we form the correct `Name*`.

Anyway, long story short there is no reason why we have to write this as concatnation, because we should be able to just `UnownedTerminatedStringSlice("?:")`.

Reply:
Actually just looking at TokenType makes that clear. Will fix.